pkg/bmc/ibmc.go

```diff
@@ -0,0 +1,101 @@
+package bmc
+
+import (
+	"net/url"
+	"strings"
+)
+
+func init() {
+	registerFactory("ibmc", newIbmcAccessDetails)
+	registerFactory("ibmc+http", newIbmcAccessDetails)
+	registerFactory("ibmc+https", newIbmcAccessDetails)
+}
+
+func newIbmcAccessDetails(parsedURL *url.URL, disableCertificateVerification bool) (AccessDetails, error) {
+	return &ibmcAccessDetails{
+		bmcType:                        parsedURL.Scheme,
+		host:                           parsedURL.Host,
+		path:                           parsedURL.Path,
+		disableCertificateVerification: disableCertificateVerification,
+	}, nil
+}
+
+type ibmcAccessDetails struct {
+	bmcType                        string
+	host                           string
+	path                           string
+	disableCertificateVerification bool
+}
+
+func (a *ibmcAccessDetails) Type() string {
+	return a.bmcType
+}
+
+// NeedsMAC returns true when the host is going to need a separate
+// port created rather than having it discovered.
+func (a *ibmcAccessDetails) NeedsMAC() bool {
+	// For the inspection to work, we need a MAC address
+	// https://github.com/metal3-io/baremetal-operator/pull/284#discussion_r317579040
+	return true
+}
+
+func (a *ibmcAccessDetails) Driver() string {
+	return "ibmc"
+}
+
+func (a *ibmcAccessDetails) DisableCertificateVerification() bool {
+	return a.disableCertificateVerification
+}
+
+const ibmcDefaultScheme = "https"
+
+// DriverInfo returns a data structure to pass as the DriverInfo
+// parameter when creating a node in Ironic. The structure is
+// pre-populated with the access information, and the caller is
+// expected to add any other information that might be needed (such as
+// the kernel and ramdisk locations).
+func (a *ibmcAccessDetails) DriverInfo(bmcCreds Credentials) map[string]interface{} {
+
+	ibmcAddress := []string{}
+	schemes := strings.Split(a.bmcType, "+")
+	if len(schemes) > 1 {
+		ibmcAddress = append(ibmcAddress, schemes[1])
+	} else {
+		ibmcAddress = append(ibmcAddress, ibmcDefaultScheme)
+	}
+	ibmcAddress = append(ibmcAddress, "://")
+	ibmcAddress = append(ibmcAddress, a.host)
```

Comment:
This method of building a URL is not going to work correctly if a.host is an IPv6 address, because the address needs to be wrapped in []. Please use a net/url.URL type to construct a URL string. https://golang.org/pkg/net/url/#URL

Reply:
a.host should have wrapped by [] already. check testcase: https://github.com/metal3-io/baremetal-operator/pull/530/commits/ba75298a5c3e6445cbc2501c49314389efb8a7f2#diff-e2892b3ea0460362bf37f0f558c6fd68R297